### DISCUSSION TITLE
How to mount a smb share to container before run my application

### DISCUSSION THREAD

**Discussion by dameng324** [dotnet/sdk-container-builds#575] [Q&A] [answered]
My application relies on some data via smb share. 
So I want to mount a smb share to filesystem in container before run my application.
I have created a aspnet runtime base image with cifs installed. 
But how to change the image's entrypoint to `mount -t cifs -o username=user,password=password //192.168.1.100/shared /mnt/smbshare && dotnet MyApplication.dll` ? Does [ContainerAppCommand](https://learn.microsoft.com/zh-cn/dotnet/core/docker/publish-as-container?pivots=dotnet-8-0#containerappcommand) and [ContainerAppCommandArgs](https://learn.microsoft.com/zh-cn/dotnet/core/docker/publish-as-container?pivots=dotnet-8-0#containerappcommandargs) will work for me?

or is there any other workaround?

**Comment by baronfel** [comment, marked as answer]
For this scenario I'd suggest using a custom Entrypoint script to handle any of the pre-startup configuration. We have an example of what this looks like here: https://github.com/baronfel/sdk-container-demo/tree/main/src%2Fcustom-entrypoint

In general, you'd use the MSBuild items you mentioned for small adjustments to the applications startup arguments, and if you had specific infrastructure/startup logic of any real size then you'd switch to a custom Entrypoint script.

**Reply by dameng324** [reply]
Thanks. I'll try it.
